core/src/main/java/org/jdbi/v3/core/qualifier/QualifiedType.java

```diff
@@ -93,6 +105,13 @@ public static QualifiedType of(GenericType<?> type, Annotation... qualifiers) {
         return QualifiedType.of(type.getType(), qualifiers);
     }
 
+    /**
+     * Creates a QualifiedType for {@code type} with the given qualifiers.
+     */
+    public static QualifiedType of(GenericType<?> type, Class<? extends Annotation>... qualifiers) {
```

Comment:
having so many overloads to squarely combine Type and GenericType with the various forms of Annotation feels pretty bloated. Wouldn't a style like `QualifiedType.of(type).with(annotations)` with a copy-setter be nicer?

Reply:
:thinking:

Reply:
Done